### PR TITLE
Update to node 20 / npm 10 / v2.0.5

### PR DESCRIPTION
**Pull request by kaitlinbolling**
#237 

- Updates to node 20 / npm 10
- Updates packages
- Updates to use Alley's shared eslint & styleling configs

### CODE REVIEW

File: client/src/components/byline-autocomplete/index.jsx [hunk 1]
```diff
@@ -41,17 +41,17 @@ const BylineAutocomplete = ({
...
     },
   };
 
   useEffect(() => {
-    if ('' !== debouncedSearchString) {
+    if (debouncedSearchString !== '') {
       doProfileSearch(debouncedSearchString);
     }
-  }, [debouncedSearchString]);
+  }, [debouncedSearchString]); // eslint-disable-line react-hooks/exhaustive-deps
```
Comment (efuller) [inline client/src/components/byline-autocomplete/index.jsx]: Wonder if we could wrap `doProfileSearch` in a `useCallback` and then we wouldn't need to `eslint-disable-line`?

File: client/src/components/byline-profiles/index.js [hunk 2]
```diff
@@ -126,17 +126,17 @@ const BylineProfiles = ({
...
     },
   };
 
   useEffect(() => {
-    if ('' !== debouncedSearchString) {
+    if (debouncedSearchString !== '') {
       doProfileSearch(debouncedSearchString);
     }
-  }, [debouncedSearchString]);
+  }, [debouncedSearchString]); // eslint-disable-line react-hooks/exhaustive-deps
```
Comment (efuller) [inline client/src/components/byline-profiles/index.js]: Wonder if we could wrap `doProfileSearch` in a `useCallback` and then we wouldn't need to `eslint-disable-line`?

File: client/src/containers/container/index.jsx [hunk 3]
```diff
@@ -28,10 +28,10 @@ const BylineSlotContainer = ({
    * The redux store schema and the meta schema are different.
    */
   useEffect(() => {
-    if (null !== profiles) {
+    if (profiles !== null) {
       saveByline(profiles);
     }
-  }, [profiles]);
+  }, [profiles]); // eslint-disable-line react-hooks/exhaustive-deps
```
Comment (efuller) [inline client/src/containers/container/index.jsx]: Wonder if we could wrap `saveByline` in a `useCallback` and then we wouldn't need to `eslint-disable-line`?

File: client/src/components/user-link-metabox/index.js [hunk 4]
```diff
@@ -34,21 +34,21 @@ const UserLinkMetaBox = ({
...
     },
   };
 
   useEffect(() => {
-    if ('' !== debouncedSearchString) {
+    if (debouncedSearchString !== '') {
       doUserSearch(debouncedSearchString);
     }
-  }, [debouncedSearchString]);
+  }, [debouncedSearchString]); // eslint-disable-line react-hooks/exhaustive-deps
```
Comment (efuller) [inline client/src/components/user-link-metabox/index.js]: Wonder if we could wrap `doUserSearch` in a `useCallback` and then we wouldn't need to `eslint-disable-line`?

File: client/src/components/user-link-metabox/index.js [hunk 5]
```diff
@@ -34,21 +34,21 @@ const UserLinkMetaBox = ({
...
 
   useEffect(() => {
     setUser(rawUser);
-  }, []);
+  }, []); // eslint-disable-line react-hooks/exhaustive-deps
```
Comment (efuller) [inline client/src/components/user-link-metabox/index.js]: Would adding `[setUser, rawUser]` as dependencies help here?

**Review by efuller — APPROVED** [review]
⚡ This looks good! Left a couple of possible suggestions in regards to `exhaustive-deps` ESLint stuff, but nothing blocking.

**Comment by kaitlinbolling** [comment]
Thanks @efuller! I added some `useCallback()` hooks and removed the ignores. Can you give this another review? Thanks!

**Review by efuller — APPROVED** [review]
⚡ looks great!

**Pull request merged**